gnmi/README.md

```diff
@@ -1,22 +1,37 @@
-# Fake gNMI Service (central datastore)
+# gNMI Service Reference Implementation (central datastore)
 
-`gnmit.GNMIServer` serves as a central datastore to which subscribing
-"tasks/agents" can be attached.
+`gnmi.GNMIServer` serves as the reference implementation of gNMI for lemming and the central datastore to which reconcilers can be attached.
 
-There are two ways to subscribe to the central datastore:
+## External Clients
 
-## 1 Subscribing using a `gnmit.Task` interface
+External clients use gNMI to configure and read state of the reference device.
+## Internal Reconcilers
 
-To use this method, define a function that satisfies the `gnmit.TaskRoutine`
-function signature, and register the task in [lemming.go](../lemming.go).
+Lemming is designed as an eventually consistent system where the gNMI cache is used as the central datastore. Reconcilers are attached to the gNMI cache and  are responsible reading the intended config and reconciling state.
+Reconcilers can optionally validate incoming SetRequest to prevent semantically incorrect values from being applied. For example, the interface reconciler can validate that MTU > 64 for Ethernet interfaces.
```

Comment:
Provide a link to the reconciler interface. Or to prevent broken link can copy the interface here (it's ok if it becomes a bit out-of-date later).

Reply:
linked pkg.go.dev link as that will be always up to date